platform/admin/admin-configuration-configinstaller/src/main/java/org/codice/ddf/admin/configuration/ConfigurationInstaller.java

```diff
@@ -0,0 +1,201 @@
+ * <td>Y</td><td>X</td><td>Felix file prop updated, delete config file and update map</td>
+ * </tr>
+ * <tr>
+ * <td>Y</td><td>Y</td><td>Felix file prop did not change, do nothing</td>
+ * </tr>
+ * </table>
+ *
+ * @see #handleUpdate(FelixConfig)
+ * @see #handleDelete(String)
+ */
+public class ConfigurationInstaller implements SynchronousConfigurationListener {
+    private static final Logger LOGGER = LoggerFactory.getLogger(ConfigurationInstaller.class);
+
+    private final ConfigurationAdmin configurationAdmin;
+
+    private final ConfiguratorFactory configuratorFactory;
+
+    private final ConfigActions configActions;
+
+    private Map<String, File> pidFileMap;
+
+    public ConfigurationInstaller(ConfigurationAdmin configurationAdmin,
+            ConfiguratorFactory configuratorFactory, ConfigActions configActions) {
+        this.configurationAdmin = configurationAdmin;
+        this.pidFileMap = Collections.emptyMap();
+
+        this.configuratorFactory = configuratorFactory;
+        this.configActions = configActions;
+    }
+
+    /**
+     * @return a read-only map of configuration pids to felix file install properties.
+     */
+    public Map<String, File> getPidFileMap() {
+        return Collections.unmodifiableMap(pidFileMap);
+    }
+
+    public void init() throws IOException, InvalidSyntaxException {
+        Configuration[] configs = configurationAdmin.listConfigurations(null);
+        if (configs == null) {
+            return;
+        }
+        this.pidFileMap = Arrays.stream(configs)
+                .map(FelixConfig::new)
+                .filter(config -> config.getFile() != null)
+                .collect(Collectors.toMap(FelixConfig::getPid, FelixConfig::getFile));
+    }
+
+    @Override
+    public void configurationEvent(ConfigurationEvent configurationEvent) {
+        switch (configurationEvent.getType()) {
+        case CM_UPDATED:
+            LOGGER.debug("Handling update");
+            Configuration configuration = getConfiguration(configurationEvent);
+            if (configuration == null) {
+                break;
+            }
+            handleUpdate(new FelixConfig(configuration));
+            break;
+        case CM_DELETED:
+            LOGGER.debug("Handling delete");
+            handleDelete(configurationEvent.getPid());
+            break;
+        default:
+            LOGGER.debug("Unknown configuration event type, taking no action");
+        }
+    }
+
+    /**
+     * Sync the config files in etc when an update occurs.
+     *
+     * @param config the {@link Configuration} wrapped in a {@link FelixConfig} that arrived as an event.
+     */
+    private void handleUpdate(FelixConfig config) {
+        String pid = config.getPid();
+        File felixFileNew = config.getFile();
+        File felixFileOriginal = pidFileMap.get(pid);
+
+        boolean filePropChanged = config.filePropChanged(felixFileOriginal);
+
+        if (felixFileOriginal == null && felixFileNew == null) {
```

Comment:
❗️ You should move this test above the call to `filePropChanged`. There's no reason to invoke that method if these values are both null.